extensions/common/iam/identity-trust/identity-trust-sts/identity-trust-sts-core/src/test/java/org/eclipse/edc/iam/identitytrust/sts/defaults/StsClientTokenIssuanceIntegrationTest.java

```diff
@@ -72,10 +73,8 @@ void setup() {
         privateKeyResolver = new VaultPrivateKeyResolver(keyParserRegistry, vault, mock(), mock());
 
         tokenGeneratorService = new StsClientTokenGeneratorServiceImpl(
-                client -> new JwtGenerationService(keyId -> {
-                    var pk = privateKeyResolver.resolvePrivateKey(keyId).orElse(null);
-                    return CryptoConverter.createSignerFor(pk);
-                }),
+                client -> new JwtGenerationService(keyId -> privateKeyResolver.resolvePrivateKey(keyId)
+                        .compose(pk -> Result.ofThrowable(() -> CryptoConverter.createSignerFor(pk)))),
```

Comment:
this is the same as the one instantiated in the `TokenServicesExtension`, it could be extracted in a separate class to avoid the duplication

Reply:
ah, clicked "merge" too fast. will do this in another PR down the line.